src/Workspaces/Core/Portable/Workspace/Workspace_Editor.cs

```diff
@@ -22,6 +22,26 @@ public abstract partial class Workspace
         private readonly Dictionary<DocumentId, TextTracker> _textTrackers = new Dictionary<DocumentId, TextTracker>();
 
         /// <summary>
+        /// Unloading projects takes a serialization lock, which can cause Shared Project
+        /// IVsHierarchys to notify us that their context hierarchy has changed. In response to
```

Comment:
i don't understand the very first statement.  how is taking a serialization lock causing 'Shared Project Hierarchys to notify us that their context hierarchy has changed'?